RecoEgamma/EgammaTools/interface/GainSwitchTools.h

```diff
@@ -60,7 +60,7 @@ class GainSwitchTools {
 
 template<bool noZS>
 reco::GsfElectron::ShowerShape 
-GainSwitchTools::redoEcalShowerShape(reco::GsfElectron::ShowerShape showerShape,const reco::SuperClusterRef& superClus, const EcalRecHitCollection* recHits,const CaloTopology* topology,const CaloGeometry* geometry)
```

Comment:
Thanks for fixing the HCAL showershapes. However this function was only meant to apply to the ECAL ones. The HCAL ones are more complicated as 

hcalDepth1OverEcal and hcalDepth2OverEcal (standard showershapes)  use the regression ecal energy and will be automatically corrected here for electrons: http://cmslxr.fnal.gov/source/DataFormats/EgammaCandidates/src/GsfElectron.cc?v=CMSSW_8_0_22

So we cant correct here.